src/components/routes/qdm/TestCaseRoutes.tsx

```diff
@@ -28,12 +31,52 @@ const TestCaseRoutes = () => {
     };
   }, []);
 
-  // Setup a context provider which holds measure state, cqmMeasure and valueSets similar to QiCore/TestCaseRoutes
+  useEffect(() => {
+    if (measure) {
+      if (measure.cqlErrors || !measure.elmJson) {
+        setErrors((prevState) => [
+          ...prevState,
+          "An error exists with the measure CQL, please review the CQL Editor tab.",
+        ]);
+      }
+      if (!measure?.groups?.length) {
+        setErrors((prevState) => [
+          ...prevState,
+          "No Population Criteria is associated with this measure. Please review the Population Criteria tab.",
+        ]);
+      }
+
+      if (!errors?.length) {
+        cqmService.current
+          .convertToCqmMeasure(measure)
+          .then((convertedMeasure) => {
+            setCqmMeasure(convertedMeasure);
+          })
+          .catch((err) => {
+            setErrors((prevState) => [...prevState, err.message]);
+          });
+      }
+    }
+  }, [measure]);
+
+  useEffect(() => {
+    if (cqmMeasure) {
+      terminologyService.current
```

Comment:
Maybe just do this `setValueSets([])` instead of adding incomplete implementation and with no testcases causing coverage drop. I think we have story to implment this.

Reply:
Thank you, I have changed the code